zerver/lib/user_groups.py

```diff
@@ -185,35 +186,51 @@ def access_user_group_for_deactivation(
             **{f"{setting_name}__in": setting_group_ids_using_deactivating_user_group}
         )
 
-    if (
-        Stream.objects.filter(realm_id=user_group.realm_id, deactivated=False)
-        .filter(stream_setting_query)
-        .exists()
+    for stream in Stream.objects.filter(realm_id=user_group.realm_id, deactivated=False).filter(
+        stream_setting_query
     ):
-        raise JsonableError(_("You cannot deactivate a user group which is used for setting."))
+        objection_settings = [
+            setting_name
+            for setting_name in Stream.stream_permission_group_settings
+            if getattr(stream, setting_name + "_id")
+            in setting_group_ids_using_deactivating_user_group
+        ]
+        if len(objection_settings) > 0:
+            objections.append(dict(type="stream", stream_id=stream.id, settings=objection_settings))
```

Comment:
Do we want to use `channel` in type here? I see we still use `stream` for variables.

Reply:
I think when adding something new like this to the API (in this case, an error response format), we decided that using `channel` is desirable. We've not attempted to bulk-migrate the server to use `channel`.

Reply:
Updated the type to `channel` but the variables, both in server and webapp, still use stream.